rust/libceed/src/lib.rs

```diff
@@ -132,11 +134,24 @@ pub enum EvalMode {
     Curl = bind_ceed::CeedEvalMode_CEED_EVAL_CURL as isize,
     Weight = bind_ceed::CeedEvalMode_CEED_EVAL_WEIGHT as isize,
 }
+impl EvalMode {
+    pub(crate) fn from_u32(value: u32) -> EvalMode {
+        match value {
+            bind_ceed::CeedEvalMode_CEED_EVAL_NONE => EvalMode::None,
+            bind_ceed::CeedEvalMode_CEED_EVAL_INTERP => EvalMode::Interp,
+            bind_ceed::CeedEvalMode_CEED_EVAL_GRAD => EvalMode::Grad,
+            bind_ceed::CeedEvalMode_CEED_EVAL_DIV => EvalMode::Div,
+            bind_ceed::CeedEvalMode_CEED_EVAL_CURL => EvalMode::Curl,
+            bind_ceed::CeedEvalMode_CEED_EVAL_WEIGHT => EvalMode::Weight,
+            _ => panic!("Unknown value: {}", value),
+        }
+    }
+}
 
 // -----------------------------------------------------------------------------
 // Ceed error
 // -----------------------------------------------------------------------------
-type Result<T> = std::result::Result<T, CeedError>;
+pub type Result<T> = std::result::Result<T, CeedError>;
 
 #[derive(Debug)]
 pub struct CeedError {
```

Comment:
We should probably just call this an `Error`, as it is already `libceed::[Ceed]Error`

Reply:
Agreed.